### ISSUE TITLE
Permission denied /tmp/packagedoc_config.json on OSD

### ISSUE THREAD

**Issue by ngoan1608** [test-fullautomation/python-genpackagedoc#54]
Hi,

I faced below issue when trying to executing aio-test-trigger with `sudo` permission

- Test case:
  `test_GenPackageDoc.py::Test_GenPackageDoc::test_GenPackageDoc_1`
- Error:
```
Exception: !!! EXCEPTION !!!
[CPackageDocConfig.__init__] : !!! EXCEPTION !!!
[CFile.Write] : !!! EXCEPTION !!!
[CFile.__OpenForWriting] : Not possible to open file '/tmp/packagedoc_config.json' for writing.
Reason: [Errno 13] Permission denied: '/tmp/packagedoc_config.json'!
```
- Permission of that file:
`-rw-rw-r-- 1 gitlab-runner gitlab-runner 2.0K Jun  4 23:21 /tmp/packagedoc_config.json`

It seems that the error comes when that file was created before by other user (1st user) then executing with different user (2nd user).
2nd user does not have `write` permission to that file and cause error when processing genpackagedoc.

Proposals:
- Implement to allow `write` permission for that file to all users.
- Change location of packagedoc_config.json to user directory (`/tmp` is not good place for user's stuff)

Thank you,
Ngoan

**Comment by test-fullautomation** [comment]
Hi Ngoan,
your proposals read reasonable for me.
Thank you,
 Thomas

**Comment by HolQue** [comment]
I am confused now, that suddenly such kind of problems occur. I thought to be on the save side when using the TMP folder for such kind of temporary files. It's strange that also in this folder access rights cause problems.

I can for sure also use the user folder instead of TMP, but I am not happy about this. Why using the user folder for temporary files?

And will the usage of the users folder prevent us from the observed access right problems?

Might it be a solution, when I delete the temporary file at end, when I do not need the file any more? In this case the process, who deletes the file is the same process who create the file. This should not cause problems. And the next process starts from scratch (with creating and deleting the file).

What do you think?

**Comment by test-fullautomation** [comment]
Hi Holger,
as discussed, deleting the file in addition at the end of the test is a good choice.
Thank you,
 Thomas

**Comment by HolQue** [comment]
Should be solved with:
https://github.com/test-fullautomation/python-genpackagedoc/pull/55

**Comment by test-fullautomation** [comment]
solved and already merge with #55
